analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DebuggerDisplayUsesExistingMembers.cs

```diff
@@ -0,0 +1,247 @@
+    [DebuggerDisplay("{SomeProperty ,nq}")]
+    [DebuggerDisplay("{SomeProperty, nq}")]
+    [DebuggerDisplay("{SomeProperty,nq }")]
+    [DebuggerDisplay("{Nonexistent,nq}")]  // Noncompliant
+    [DebuggerDisplay("{Nonexistent ,nq}")] // Noncompliant
+    [DebuggerDisplay("{Nonexistent, nq}")] // Noncompliant
+    [DebuggerDisplay("{Nonexistent,nq }")] // Noncompliant
+    int SomeProperty => 1;
+}
+
+public class SupportOptionalAttributeParameter
+{
+    [DebuggerDisplay("{SomeProperty}", Name = "Any name")]
+    [DebuggerDisplay("{Nonexistent}", Name = "Any name")]                                                     // Noncompliant {{'Nonexistent' doesn't exist in this context.}}
+    //               ^^^^^^^^^^^^^^^
+    [DebuggerDisplay("{Nonexistent}", Name = "Any name", Type = nameof(SupportOptionalAttributeParameter))]   // Noncompliant {{'Nonexistent' doesn't exist in this context.}}
+    //               ^^^^^^^^^^^^^^^
+    [DebuggerDisplay("{Nonexistent}", Name = "Any name", Target = typeof(SupportOptionalAttributeParameter))] // Noncompliant {{'Nonexistent' doesn't exist in this context.}}
+    //               ^^^^^^^^^^^^^^^
+    int SomeProperty => 1;
+}
+
+public class SupportInheritance
+{
+    public class BaseClass
+    {
+        int SomeProperty => 1;
+    }
+
+    public class SubClass : BaseClass
+    {
+        [DebuggerDisplay("{SomeProperty}")] // Compliant, defined in base class
+        int OtherProperty => 1;
+    }
+}
+
+public class SupportAccessModifiers
+{
+    public class BaseClass
+    {
+        public int PublicProperty => 1;
+        internal int InternalProperty => 1;
+        protected int ProtectedProperty => 1;
+        private int PrivateProperty => 1;
+
+        [DebuggerDisplay("{PublicProperty}")]
+        [DebuggerDisplay("{InternalProperty}")]
+        [DebuggerDisplay("{ProtectedProperty}")]
+        [DebuggerDisplay("{PrivateProperty}")]
+        int SomeProperty => 1;
+    }
+
+    public class SubClass : BaseClass
+    {
+        [DebuggerDisplay("{PublicProperty}")]
+        [DebuggerDisplay("{InternalProperty}")]
+        [DebuggerDisplay("{ProtectedProperty}")]
+        [DebuggerDisplay("{PrivateProperty}")]
+        int OtherProperty => 1;
+    }
+}
+
+public class SupportAttributeTargets
+{
+    [assembly: DebuggerDisplay("{Nonexistent}")] // Noncompliant, attribute at assembly level, referencing a non-existing member
+    [field: DebuggerDisplay("{Nonexistent}")]    // Noncompliant, attribute ignored, still referencing a non-existing member
+    [property: DebuggerDisplay("{Nonexistent}")] // Noncompliant, attribute taken into account
+    int SomeProperty => 1;
+}
+
+namespace WithTypeAlias
+{
+    using DebuggerDisplayAlias = System.Diagnostics.DebuggerDisplayAttribute;
+
+    public class Test
+    {
+        [DebuggerDisplayAlias("{Nonexistent}")] int WithAlias => 1; // Noncompliant: attribute name also checked at semantic level
+    }
+}
```

Comment:
It might be beneficial to add some compiler error scenarios to the test cases. Just to see that it doesn't break the analyzer.

`[DebuggerDisplayAttribute] int WithSuffix => 1; // Error - missing attribute parameter`